File: 0938-range-sum-of-bst/0938-range-sum-of-bst.cpp
#include <stack>
class Solution {
public:
    int rangeSumBST(TreeNode* root, int low, int high) {
        std::stack<TreeNode*> s;
        int sum = 0;
        s.push(root);
        while (!s.empty()) {
            TreeNode* curr = s.top();
            s.pop();
            if (curr->val >= low && curr->val <= high) {
                sum+= curr->val;
            }
            if (curr->left != nullptr) {
                s.push(curr->left);
            }
            if (curr->right != nullptr) {
                s.push(curr->right);
            }
        }
        return sum;
    }
};

File: 0938-range-sum-of-bst/0938-range-sum-of-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int rangeSumBST(TreeNode root, int low, int high) {
        int[] res = new int[1];
        solve(root,low,high,res);
        return res[0];
    }

   public void solve(TreeNode root, int low, int high, int[] sum){
        if (root == null){
            return;
        }
        if (root.val > high){
            solve(root.left,low,high,sum);
        }else if (root.val < low){
            solve(root.right,low,high,sum);
        }else{
            sum[0] += root.val;
            solve(root.right,low,high,sum);
            solve(root.left,low,high,sum);
        }
    }
}